gemma-core/src/main/java/ubic/gemma/core/util/AbstractCLI.java

```diff
@@ -107,24 +121,42 @@
     protected String host = DEFAULT_HOST;
     private CommandLine commandLine;
 
-    public AbstractCLI() {
-        this.buildStandardOptions();
-        this.buildOptions();
+    /**
+     * Run a command. If
+     *
+     * @param args arguments
+     * @return exit code
+     */
+    public int executeCommand( String[] args ) {
+        StopWatch watch = new StopWatch();
+        watch.start();
+        try {
+            buildOptions();
+            buildStandardOptions();
+            processCommandLine( args );
+            doWork();
+            return errorObjects.isEmpty() ? SUCCESS : FAILURE_FROM_ERROR_OBJECTS;
+        } catch ( Exception e ) {
+            log.error( getCommandName() + " failed.", e );
+            return FAILURE;
+        } finally {
+            resetLogging();
+            AbstractCLI.log.info( "Elapsed time: " + watch.getTime() + " ms." );
```

Comment:
let's leave timing information in seconds, not ms, for human readable log messages

Reply:
Both were used, but it's okay to use seconds.